src/MFPEncoder.ts

```diff
@@ -105,83 +111,114 @@ export default class MFPBaseEncoder {
-    printResolution: number,
-    customizer: BaseCustomizer,
-  ): Promise<MFPLayer[] | MFPLayer | null> {
-    const layer = layerState.layer as VectorTileLayer;
-    const {MVTEncoder} = await import('@geoblocks/print');
-    const encoder = new MVTEncoder();
-    const printExtent = customizer.getPrintExtent();
-    const width = getExtentWidth(printExtent) / printResolution;
-    const height = getExtentHeight(printExtent) / printResolution;
-    const canvasSize: [number, number] = [width, height];
-    const printOptions = {
-      layer,
-      printExtent: customizer.getPrintExtent(),
-      tileResolution: printResolution,
-      styleResolution: printResolution,
-      canvasSize: canvasSize,
+  addRenderAsSVG(layerState: State, encoded: MFPVectorLayer) {
+    const renderAsSvg = layerState.layer.get('renderAsSvg');
+    if (renderAsSvg !== undefined) {
+      encoded.renderAsSvg = renderAsSvg;
+    }
+  }
+
+  /**
+   * @returns An Encoded WMS Image layer from an Image Layer (high level method).
+   */
+  encodeImageLayerState(layerState: State, customizer: BaseCustomizer): MFPWmsLayer | null {
+    const layer = layerState.layer;
+    if (!(layer instanceof ImageLayer)) {
+      console.assert(layer instanceof ImageLayer);
+    }
+    const source = layer.getSource();
+    if (source instanceof ImageWMSSource) {
+      return this.encodeImageWmsLayerState(layerState, customizer);
+    }
+    return null;
+  }
+
+  /**
+   * @returns An Encoded WMS Image layer from an Image WMS Source (high level method).
+   */
+  encodeImageWmsLayerState(layerState: State, customizer: BaseCustomizer) {
+    const layer = layerState.layer;
+    const source = layer.getSource() as ImageWMSSource;
+    console.assert(layer instanceof ImageWMSSource);
+    const url = source.getUrl();
+    if (url !== undefined) {
+      return this.encodeWmsLayerState(layerState, url, source.getParams(), customizer);
+    }
+    return null;
+  }
+
+  /**
+   * @returns An Encoded WMS Image layer from an Image WMS Source.
+   */
+  encodeWmsLayerState(layerState: State, url: string, params: any, customizer: BaseCustomizer): MFPWmsLayer {
+    const layer = layerState.layer;
+
+    if (url.startsWith('//')) {
+      url = window.location.protocol + url;
+    }
+    const url_object = new URL(url);
+
+    let serverType = undefined;
+    if (params.SERVERTYPE !== 'arcgis') {
+      serverType = params.SERVERTYPE;
+    }
+
+    const layers = params.LAYERS instanceof String ? params.LAYERS.split(',') : params.LAYERS;
+    const styles = params.STYLES instanceof String ? params.STYLES.split(',') : params.STYLES;
+
+    return {
+      name: layer.get('name'),
+      baseURL: getAbsoluteUrl(url_object.origin + url_object.pathname),
+      imageFormat: 'FORMAT' in params ? params.FORMAT : 'image/png',
+      layers,
+      customParams: {},
+      serverType,
+      type: 'wms',
+      opacity: layer.getOpacity(),
+      version: params.VERSION,
+      useNativeAngle: true,
```

Comment:
What is native angle?

Reply:
It's a print option that indicates that we will ask the WMS server to rotate, otherwise the print will ask for a wider image and do the rotation itself.

Reply:
no clue